src/app/shared-pages/categories/category-form/category-form.component.ts

```diff
@@ -10,12 +11,30 @@ export class CategoryFormComponent implements OnInit {
 
   @Input() category: Category = {};
   @Output() onSubmit: EventEmitter<Category> = new EventEmitter<Category>();
+  categoryFormGroup: FormGroup;
 
-  constructor() { }
+  constructor(
+    private formBuilder: FormBuilder,
+  ) { }
 
-  ngOnInit() {}
+  ngOnInit() {
+    const activeCategory = this.category.hasOwnProperty('active') ? this.category.active : false;
+    this.categoryFormGroup = this.formBuilder.group({
+      name: [this.category.name, Validators.required],
+      percentage: [this.category.percentage, Validators.compose([
+        Validators.required,
+        Validators.min(0),
+        Validators.max(100),
+      ])],
+      active: [activeCategory, Validators.required],
```

Comment:
solo algo chiquito. `!!this.category.active` y eliminar `const activeCategory`

Reply:
chever primo, ya esta